### PR TITLE
Segment Tree Range Minimum Query in  C

### LINKED ISSUE #2164: Segment Tree Range Minimum Query
- [x] C - @MdAkdas
- [x] C++
- [ ] C#
- [ ] CoffeeScript
- [x] Dart - @Rukmini-Meda
- [ ] Go
- [x] Java
- [ ] JS - @JanaSabuj
- [ ] Kotlin
- [ ] Php
- [x] Python
- [x] Readme - @chandrikadeb7
- [ ] Ruby
- [ ] Rust
- [ ] TypeScript

### PR DESCRIPTION
**Pull request by MdAkdas**
Fixes #2164

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [ ] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- 
- 

#### Languages Used:
- 
- 

#### Files Added:
- 
- 

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by MdAkdas** [comment]
@jainaman224 . it's been many days. please review this.

File: Segment_Tree_RMQ/Segement_Tree_RMQ.c [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+	int e = query(tree, start, mid, l, r, 2 * treenode);
+	int f = query(tree, mid + 1, end, l, r, 2 * treenode + 1);
+
+	return min(e, f);
+}
+
+int main() 
+{
+    int array[5] = {-7, 3, -1, 8, 1};
+    int size = 5;
```
Comment (asha15) [inline Segment_Tree_RMQ/Segement_Tree_RMQ.c:59]: shall we change the code to take dynamic inputs?

File: Segment_Tree_RMQ/Segement_Tree_RMQ.c [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    int* segTree = new int[4*size];
+
+	build(array, segTree, 0, size - 1, 1);
```
Comment (asha15) [inline Segment_Tree_RMQ/Segement_Tree_RMQ.c:62]: Correct the indentation

File: Segment_Tree_RMQ/Segement_Tree_RMQ.c [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+    int start = 2, end = 5;
+    int result = query(segTree, 0, size - 1, start - 1, end - 1, 1);
+
+    printf("Minimum value in index range %d to %d is: %d", start, end, result);
+
+	return 0;
```
Comment (asha15) [inline Segment_Tree_RMQ/Segement_Tree_RMQ.c:69]: correct the indentation

File: Segment_Tree_RMQ/Segement_Tree_RMQ.c [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+	mid = (start + end) / 2;
+
+    //compute the values in the left and right subtrees
+	build(a, tree, start, mid, 2 * treenode);
+	build(a, tree, mid + 1, end, 2 * treenode + 1);
+
+    /*store the minimum value in the first and
+    second half of the interval*/
+	tree[treenode] = min(tree[2 * treenode], tree[2 * treenode + 1]);
+	return;
```
Comment (asha15) [inline Segment_Tree_RMQ/Segement_Tree_RMQ.c:27]: please correct the indentation

File: Segment_Tree_RMQ/Segement_Tree_RMQ.c [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+	}
+
+	int mid;
+	mid = (start + end) / 2;
+
+    //return the minimum value
+	int e = query(tree, start, mid, l, r, 2 * treenode);
+	int f = query(tree, mid + 1, end, l, r, 2 * treenode + 1);
+
+	return min(e, f);
```
Comment (asha15) [inline Segment_Tree_RMQ/Segement_Tree_RMQ.c:53]: please correct the indentation

**Comment by asha15** [comment]
Good job!

please squash your commits , so i can merge it

**Comment by MdAkdas** [comment]
Squashed the previous commits. Please check.

**Pull request merged**